fe/fe-core/src/main/cup/sql_parser.cup

```diff
@@ -4288,6 +4293,11 @@ array_slice ::=
   {: RESULT = new ArraySliceExpr(e, new IntLiteral(0), new IntLiteral(-1)); :}
   ;
 
+arrow_expr ::=
```

Comment:
Is this arrow_expr neceesary for JOSN?
I'm a little worried if this definition  will be conflict with lambda expression?

Reply:
The arrow expression looks like '->', currently used a syntax sugar for JSON. 

Current implementation of arrow expression is just translate into `json_query($value, $path)`, which would conflict with some other usage like lambda expression.

However, I think it's quite easy to avoid conflicts, just look up the context of this expression, and translate it according to the context.

Anyway, some existing jsonpath sugar, like `col->'field'`, `col.k1.k2`, `col[k1][k2]`, are always has ambiguity with other SQL syntax, there's no simple approach to create an unique syntax.

Reply:
What I'm worried is that these two expr will cause shift/reduce conflict when parsing phase if we support lambda later.
One way is to create a super expr compatible with both ArorwExpr and Lambda.
Is it possible?

Reply:
One possible way: 

arrow_expr is defined as : `left_expr '->' right_expr`.


Resolve rule: 
1. If `left_expr` could be resolved a `json` type, including column_ref or function result, it's a json access expression. And the `right_expr` must be a literal string expression.
2. Otherwise, `left_expr` could not be resolved, the whole expression is recognized as **lambda expression**

Which means, in parsing phase they could not be differentiated, only analyzer could tell the truth.